src/inventorymanager.cpp

```diff
@@ -533,20 +533,42 @@ void IMoveAction::clientApply(InventoryManager *mgr, IGameDef *gamedef)
 	// Optional InventoryAction operation that is run on the client
 	// to make lag less apparent.
 
-	Inventory *inv_from = mgr->getInventory(from_inv);
-	Inventory *inv_to = mgr->getInventory(to_inv);
-	if(!inv_from || !inv_to)
+	/*
+		Disable moving items out of craftpreview
+	*/
+	if (from_list == "craftpreview")
```

Comment:
No, this is low-level code, hardcoding behaviour here is horrible style. First you introduce a system for inventory movement prediction control, and then you hardcode it here?